DynamicSizePool.hpp

```diff
@@ -24,8 +27,8 @@ class DynamicPoolAllocator
   };
 
   // Allocator for the underlying data
-  typedef FixedPoolAllocator<struct Block, IA, (1<<6)> BlockAlloc;
-  BlockAlloc blockAllocator;
+  typedef FixedSizePool<struct Block, IA, IA, (1<<6)> BlockPool;
```

Comment:
I'm not sure I understand this. Shouldn't this be "MA, IA" instead of "IA, IA"?

Reply:
The new FixedPool can use different allocators for the meta-data and the actual data (e.g. for storing data in GPU memory). In this case, we want to use the same allocator for both. So I think this is correct.

Reply:
I'm not sure I see why FixedPool has use cases for separate allocators if meta-data and actual data and DynamicSizePool does not?  What is the unique use case(s) that FixedPool has that DynamicSizePool does not with respect to this?

Reply:
FixedPool was storing the metadata and actual data all in the same allocation. DynamicPool does not (they are already separate).

Reply:
Okay.  I see.